core/src/main/java/jenkins/widgets/HistoryPageEntry.java

```diff
@@ -59,14 +59,7 @@ protected static long getEntryId(@Nonnull Object entry) {
             return ((Queue.Item) entry).getId();
         } else if (entry instanceof Run) {
             Run run = (Run) entry;
-            long queueId = run.getQueueId();
-            if (queueId == Run.QUEUE_ID_UNKNOWN) {
-                // Backward compatibility. This is a run from before the Queue.Item IDs
-                // were mapped onto their resulting Run instance.
-                return (Integer.MIN_VALUE + run.getNumber());
-            } else {
-                return queueId;
-            }
+            return (Integer.MIN_VALUE + run.getNumber());
```

Comment:
@tfennelly Right. FWIW In testing I got some weird behavior that when I triggered several builds (all ~0ms duration) in quick succession, afterwards all "AJAX loaded items" were duplicated in the widget.

Still, I consider some minor UI weirdness (reloading fixes it) a small price to pay to get consistent ordering AND the pagination. This specific approach would also be temporary until the queueId situation is resolved.